webapi/Skills/ChatSkills/SemanticMemorySkill.cs

```diff
@@ -64,53 +65,118 @@ public class SemanticChatMemorySkill
         var remainingToken = tokenLimit;
 
         // Search for relevant memories.
-        List<MemoryQueryResult> relevantMemories = new();
-        foreach (var memoryName in this._promptOptions.MemoryMap.Keys)
+        List<(Citation Citation, Citation.Partition Memory)> relevantMemories = new();
+        foreach (var memoryName in this._promptOptions.MemoryMap.Keys.Append(this._promptOptions.DocumentMemoryName))
         {
-            string memoryCollectionName = SemanticChatMemoryExtractor.MemoryCollectionName(chatId, memoryName);
-            try
+            await SearchMemoryAsync(memoryName).ConfigureAwait(false);
+        }
+
+        var builderMemory = new StringBuilder();
+
+        if (relevantMemories.Count > 0)
+        {
+            var memoryMap = ProcessMemories();
+
+            foreach (var memoryName in this._promptOptions.MemoryMap.Keys)
             {
-                var results = textMemory.SearchAsync(
-                    memoryCollectionName,
-                    query,
-                    limit: 100,
-                    minRelevanceScore: this.CalculateRelevanceThreshold(memoryName, chatSession!.MemoryBalance));
-                await foreach (var memory in results)
+                FormatMemories(memoryName);
+            }
+
+            FormatSnippets();
+
+            void FormatMemories(string memoryName)
+            {
+                if (!memoryMap.TryGetValue(memoryName, out var memories))
+                {
+                    return;
+                }
+
+                foreach (var memory in memories)
                 {
-                    relevantMemories.Add(memory);
+                    var memoryText = $"[{memoryName}] {memory}\n";
+                    builderMemory.Append(memoryText);
                 }
             }
-            catch (SKException connectorException)
+
+            void FormatSnippets()
             {
-                // A store exception might be thrown if the collection does not exist, depending on the memory store connector.
-                this._logger.LogError(connectorException, "Cannot search collection {0}", memoryCollectionName);
+                if (!memoryMap.TryGetValue(this._promptOptions.DocumentMemoryName, out var memories))
+                {
+                    return;
+                }
+
+                foreach ((var memory, var citation) in memories)
+                {
+                    var memoryText = $"[{citation.SourceName}] {memory}\n";
+                    builderMemory.Append(memoryText);
```

Comment:
There's a manual check in PromptDialog.tsx that checks for document memories from the preamble concatenated before -- is there a way to know document memories from semantic memory now?      

maybe add a Preamble here or separate the two?          

This is the check `!(value as string).includes('User has also shared some document snippets:')`

Reply:
Thanks for this call-out.  Right...I was wondering about the document memory separation...in this change its mixed by relevance...i think.  Agree this is an unnecessary change and will address.